Makefile

```diff
@@ -312,7 +320,7 @@ verify-manifests: manifests $(YQ) ## Verify manifests update.
 	yq e -e '.metadata.annotations.containerImage == "$(OPERATOR_IMAGE)"' config/manifests/bases/authorino-operator.clusterserviceversion.yaml
 
 .PHONY: verify-bundle
-verify-bundle: bundle $(YQ) ## Verify bundle update.
+verify-bundle: $(YQ) ## Verify bundle update.
 	git diff -I'^    containerImage:' -I'^                image:' --exit-code ./bundle
 	[ -z "$$(git ls-files --other --exclude-standard --directory --no-empty-directory ./bundle)" ]
 	yq e -e '.metadata.annotations.containerImage == "$(OPERATOR_IMAGE)"' bundle/manifests/authorino-operator.clusterserviceversion.yaml
```

Comment:
We should probably include here as well:

```
	yq e -e '.spec.install.spec.deployments[1].spec.template.spec.containers[0].image == "$(DEFAULT_AUTHORINO_IMAGE)"' bundle/manifests/authorino-operator.clusterserviceversion.yaml
	yq e -e '.spec.replaces == "$(shell yq e -e '.config.replaces' $(BUILD_CONFIG_FILE))"' bundle/manifests/authorino-operator.clusterserviceversion.yaml
```

Reply:
Won't the git diff catch these ones? the other ones for OPERATOR_IMAGE are because we explicitly ignore them in the git diff and verify with yq instead. Given the replaces + authorino image are set by the bundle run just prior to this the git diff should catch right?

Reply:
About `.spec.replaces`, you're right. We still need it for the Authorino image though.

Reply:
You can test it with the [verification steps](https://github.com/Kuadrant/authorino-operator/pull/161#issuecomment-1825514169) I left. Instead of verifying the bundle with the correct `AUTHORINO_VERSION`, try a different one (only in the last step). E.g.:

```sh
make verify-bundle VERSION=0.10.0 AUTHORINO_VERSION=0.17.0 CHANNELS=stable DEFAULT_CHANNEL=stable
```

It will pass as if the bundle was valid while it should not.

Reply:
Sure, but I think it'll pass anyway with the above change because `DEFAULT_AUTHORINO_IMAGE` is not affected by `AUTHORINO_VERSION` passed at command line, only whether the file is present or not.

Reply:
Right. So maybe we can define:

```make
EXPECTED_DEFAULT_AUTHORINO_IMAGE = $(DEFAULT_REGISTRY)/$(DEFAULT_ORG)/authorino:$(AUTHORINO_IMAGE_TAG)
```

And then:

```make
verify-manifests:
	[…]
	$(YQ) ea -e 'select([.][].kind == "Deployment") | select([.][].metadata.name == "authorino-webhooks").spec.template.spec.containers[0].image | . == "$(EXPECTED_DEFAULT_AUTHORINO_IMAGE)"' config/deploy/manifests.yaml
```

and

```make
verify-bundle:
	[…]
	$(YQ) e -e '.spec.install.spec.deployments[1].spec.template.spec.containers[0].image == "$(EXPECTED_DEFAULT_AUTHORINO_IMAGE)"' bundle/manifests/authorino-operator.clusterserviceversion.yaml
```

WDYT?

Reply:
I'm not sure I see the value in this tbh - happy to put it in if you like. If the bundle command is run prior and the file is present in the release branch the git diff will catch the change anyway. This additional check will ensure that if you pass the `AUTHORINO_VERSION` variable to `verify-` commands that doesn't match the one in the file that has been generated then you fail. This would only happen if for some reason the version in the file has been generated at a different version to what you're running the verify at right? is there a case I'm missing where this is likely to happen and be a problem?

Reply:
I've added this change anyway we can remove it later if it's unnecessary I guess

Reply:
The way this goes is: first you generate the manifests (a set of params is given here), then you commit the changes, then you verify.

As for the verification itself, it goes: you generate the manifests again (supposedly with the exact same set of params used before), then you check if there's a diff. If there is a diff, then verification fails; otherwise, verification passes and the manifests are considered correct.

When checking the diff, however, it ignores all lines in the diff containing `    containerImage:` or `        image:`. I'm now struggling to see why we ignore these lines. Additionally, have we checked the indentation used here since we moved to using `yq`?

In the end, I think the purpose of verifying the manifests (both base and final bundle) is checking if the set of params used when the manifests were generated was correct.

Reply:
> When checking the diff, however, it ignores all lines in the diff containing     containerImage: or         image:. I'm now struggling to see why we ignore these lines. Additionally, have we checked the indentation used here since we moved to using yq?

IIRC this is to overcome the 'chicken and egg' issue of creating builds with the containerImages set to the commit sha build of the operator. We only know the commit sha after it has been committed, so the CSV cannot contain the commit sha. Instead we check that the `OPERATOR_IMAGE` containing the commit sha is in the CSV.

Good point on the whitespace though, it looks like it needs updating will do that now.